sim.c

```diff
@@ -770,16 +770,15 @@ BEGIN_OPERATOR(zig)
   if (gline[x + 1] == '.') {
     gline[x + 1] = This_oper_char;
     mbuffer[width * y + x + 1] |= (U8)Mark_flag_sleep;
-  } else {
-    Usz n = 256;
-    if (x < n)
-      n = x;
-    for (Usz i = 0; i < n; ++i) {
-      if (gline[x - i - 1] != '.') {
-        gline[x - i] = This_oper_char;
-        break;
+  } else if (gline[x + 1] != This_oper_char && x > 1) {
```

Comment:
Unnecessary re-read from same heap address.

Testing `x > 1` in strange place. It should be done earlier and the branches rewritten.

There is no distance limit. The distance limit is important for preventing the VM from hanging for long periods of time.